nixos/modules/services/networking/netbird.nix

```diff
@@ -645,6 +668,77 @@ in
         });
       '';
     })
+    # Setup Keys login automation
+    {
+      systemd.services = pipe cfg.clients [
+        (filterAttrs (_: client: client.login.enable))
+        (mapAttrs' (
+          _: client:
+          nameValuePair "${client.service.name}-login" {
+            after = [ "${client.service.name}.service" ] ++ client.login.systemdDependencies;
+            requires = [ "${client.service.name}.service" ] ++ client.login.systemdDependencies;
+            wantedBy = [ "${client.service.name}.service" ];
+
+            serviceConfig = {
+              Type = "oneshot";
+              RemainAfterExit = true;
+
+              User = client.user.name;
+              Group = client.user.group;
+
+              RemoveIPC = true;
+              PrivateTmp = true;
+              ProtectSystem = "strict";
+              ProtectHome = "yes";
+
+              LoadCredential = [ "setup-key:${client.login.setupKeyFile}" ];
+            };
+
+            environment.NB_SETUP_KEY_FILE = "%d/setup-key";
+            /*
+              might want to do something similar to the docker entrypoint (watching log messages) instead
+              see might want to do something similar to https://github.com/netbirdio/netbird/blob/dc30dcacce4c322502975f1f491e6774efd7e1e9/client/netbird-entrypoint.sh
+            */
+            script = ''
+              set -x
+              nb='${lib.getExe client.wrapper}'
+              keyFile="''${NB_SETUP_KEY_FILE:-"/run/credentials/${client.service.name}.service/setup-key"}"
```

Comment:
That fallback is not documented in the option and a bit surprising

Reply:
This is a [well-documented path for `LoadCredential=`](https://systemd.io/CREDENTIALS/#relevant-paths) and is pretty much static.

I think I originally used it to run the standalone script by path when developing this.

PS: actually after reading throug it I think have an error here (missing the `-login` suffix).